File: Core/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using Core.Interfaces;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity, IDtoConvertable
    {
        [Column("Order_Date")]
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        [Column("Buyer_Email")]
        public required string BuyerEmail { get; set; }

        public ShippingAddress ShippingAddress { get; set; } = null!;

        public DeliveryMethod DeliveryMethod { get; set; } = null!;

        public PaymentSummary PaymentSummary { get; set; } = null!;

        public List<OrderItem> OrderItems { get; set; } = [];
        
        [Column("Sub_Total")]
        public decimal SubTotal { get; set; }

        public decimal Discount { get; set; }

        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        [Column("Payment_Intent_Id")]
        public required string PaymentIntentId { get; set; }

        public decimal GetTotal()
        {
            return SubTotal - Discount + DeliveryMethod.Price;
        }
    }
}

File: web-app/src/app/shared/models/order.ts
export interface Order {
    orderId: number,
    orderDate: string,
    buyerEmail: string,
    shippingAddress: ShippingAddress,
    deliveryMethod: string,
    deliveryFee: number,
    paymentSummary: PaymentSummary,
    orderItems: OrderItem[],
    subTotal: number,
    discount?: number,
    status: string,
    total: number,
    paymentIntentId: string,
  }
  
  export interface ShippingAddress {
    name: string,
    line1: string,
    line12?: string,
    country: string,
    city: string,
    state: string,
    postalCode: string,
  }
  
  export interface PaymentSummary {
    last4: number,
    brand: string,
    expMonth: number,
    expYear: number,
  }
  
  export interface OrderItem {
    productId: number,
    productName: string,
    pictureUrl: string,
    price: number,
    quantity: number,
  }

  export interface OrderToCreate {
    cartId: string,
    deliveryMethodId: number,
    shippingAddress: ShippingAddress,
    paymentSummary: PaymentSummary,
    discount?: number
  }